pandas/tests/io/json/test_pandas.py

```diff
@@ -1389,6 +1389,34 @@ def test_from_json_to_json_table_dtypes(self):
         result = read_json(dfjson, orient="table")
         tm.assert_frame_equal(result, expected)
 
+    def test_to_json_from_json_columns_dtypes(self):
+        expected = DataFrame.from_dict(
+            {
+                "Integer": pd.Series([1, 2, 3], dtype="int64"),
+                "Float": pd.Series([None, 2.0, 3.0], dtype="float64"),
+                "Object": pd.Series([None, "", "c"], dtype="object"),
+                "Bool": pd.Series([True, False, True], dtype="bool"),
+                "Category": pd.Series(["a", "b", None], dtype="category"),
+                "Datetime": pd.Series(
+                    ["2020-01-01", None, "2020-01-03"], dtype="datetime64[ns]"
+                ),
+            }
+        )
+        dfjson = expected.to_json(orient="columns")
```

Comment:
While the issues are specific to `orient="columns"`, can we also parameterize over other orientations here? (since seems like a test which covers dtype conversion functionality that may not be hit as comprehensively elsewhere)

Reply:
Makes sense, I added `["split", "records", "index", "columns"]`. 
44a3935